# ![nfcore/test-datasets](docs/images/test-datasets_logo.png)
Test data to be used for automated testing with the nf-core pipelines

## Introduction

nf-core is a collection of high quality Nextflow pipelines. This repository contains various files for CI and unit testing of nf-core pipelines and infrastructure.

The principle for nf-core test data is as small as possible, as large as necessary. Always ask for guidance on the [nf-core slack](https://nf-co.re/join) before adding new test data.

## Documentation

nf-core/test-datasets comes with documentation in the `docs/` directory:

01. [Add a new  test dataset](https://github.com/nf-core/test-datasets/blob/master/docs/ADD_NEW_DATA.md)
02. [Use an existing test dataset](https://github.com/nf-core/test-datasets/blob/master/docs/USE_EXISTING_DATA.md)

## Downloading test data

Due the large number of large files in this repository for each pipeline, we highly recommend cloning only the branches you would use.

```bash
git clone <url> --single-branch --branch <pipeline/modules/branch_name>
```

To subsequently clone other branches[^1]

```bash
git remote set-branches --add origin [remote-branch]
git fetch
```

## Metagenome CI test dataset: specific information

### Publication

One wastewater dataset published by [Che et al. (2019)](https://doi.org/10.1186/s40168-019-0663-0), SRX5027541, activated sludge, SWH plant,
Illumina HiSeq 4000, Paired, WGS was used to generate the test datasets.

### Assembly and tools tested

Assembly was done with nf-core/mag pipeline (v2.1.1), default settings. The file from SPAdes Assembly SRX5027541_T1_contigs.fasta was used to test the following tools:

1. Fargene
2. Macrel
3. deepARG
4. Antismash
5. Ampir (on faa file produced by prodigal)
This resulted in a list of hits with 120 unique contig names.

### Commands to generate the test dataset

Filter the assembly fasta for the list of hits

```bash
awk '{ if ((NR>1)&&($0~/^>/)) { printf("\n%s", $0); } else if (NR==1) { printf("%s", $0); } else { printf("\t%s", $0); } }' in.fa | grep -Ff patterns.txt - | tr "\t" "\n" > out.fa
```

Filter for all reads except list of hits

```bash
awk '{ if ((NR>1)&&($0~/^>/)) { printf("\n%s", $0); } else if (NR==1) { printf("%s", $0); } else { printf("\t%s", $0); } }' in.fa | grep -v patterns.txt - | tr "\t" "\n" > out.fa
```

Get a subsample of a fasta file (specify number of contigs)

```bash
seqtk sample nohits.fa 1200 >sub_nohits.fa
```

Join the hits.fa and sub_nohits.fa to the dataset

1. dataset 01 contains all 120 hits plus 1200 non-hit contigs
2. dataset 02 contains only 60 hits plus 3000 non-hit contigs

In addition prokka annotated files were added for both test-datasets.

## HMM Models

For testing HMMSearch, we downloaded the following files from the Pfam database, as generally ubiquitous gene families:

- [ABC_tran](http://pfam.xfam.org/family/PF00005)
- [Pkinase](http://pfam.xfam.org/family/PF00069)

and provided a user computed hmm model for the antimicrobial peptide family of [Bacteriocins](https://raw.githubusercontent.com/nf-core/test-datasets/funcscan/hmms/mybacteriocin.hmm)


## Metagenome AWS full test dataset: specific information

Our AWS full test dataset are contigs generated by the [MGnify](https://www.ebi.ac.uk/metagenomics/) service from the ENA. We used contigs generated from assemblies of chicken cecum shotgun metagenomes (study accession: MGYS00005631).

To download the full test dataset to run offline, you can run the following commands

```bash
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575326/file/ERZ1664520_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575327/file/ERZ1664518_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575328/file/ERZ1664517_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575329/file/ERZ1664528_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575330/file/ERZ1664526_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575331/file/ERZ1664524_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575332/file/ERZ1664523_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575333/file/ERZ1664521_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575334/file/ERZ1664516_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575335/file/ERZ1664515_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575336/file/ERZ1664511_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575337/file/ERZ1664510_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575338/file/ERZ1664509_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575339/file/ERZ1664508_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575340/file/ERZ1664507_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575341/file/ERZ1664506_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575342/file/ERZ1664505_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575343/file/ERZ1664504_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575344/file/ERZ1664503_FASTA.fasta.gz
wget https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575345/file/ERZ1664501_FASTA.fasta.gz
```

## Support

For further information or help, don't hesitate to get in touch on our [Slack organisation](https://nf-co.re/join/slack) (a tool for instant messaging).

[^1]: From [stackoverflow](https://stackoverflow.com/a/60846265/11502856)
