# test-datasets: `denovotranscript`

This branch contains test data to be used for automated testing with the [nf-core/denovotranscript](https://github.com/nf-core/denovotranscript) pipeline.

## Contents of this repository

`samplesheet/samplesheet.csv`: Samplesheet for reads for minimal test dataset

`samplesheet/samplesheet_full.csv`: Samplesheet for reads for full test dataset

`data/minimal/*.fastq.gz`: Subsampled FASTQ files of RNA-seq reads containing 2 000 reads each

`data/minimal/*.fa`: Assembled transcriptome generated from the minimal test dataset

`data/full/*.fastq.gz`: Subsampled FASTQ files of RNA-seq reads containing 80 000 reads each

`data/full/*.fa`: Assembled transcriptome generated from the full test dataset

`data/Panulirus_ornatus_pep.fa`: Reference proteins to be used for Transrate step in the pipeline

## Test dataset source

Paired-end RNA-seq reads are from whole _Panulirus ornatus_ ([PRJNA664650](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA664650)). See associated reference:

> Hyde CJ, Fitzgibbon QP, Elizur A, Smith GG, Ventura T. Transcriptional profiling of spiny lobster metamorphosis reveals three new additions to the nuclear receptor superfamily. BMC Genomics. 2019 Jun 28;20(1):531. doi: 10.1186/s12864-019-5925-5.

Reference proteins for _P. ornatus_ are from [figshare](https://figshare.com/articles/dataset/The_chromosome-level_genome_of_the_long-tailed_marine-living_ornate_spiny_lobster_i_Panulirus_ornatus_i_/24654915/1). See associated reference:

> Ren X, Sun D, Lv J, Gao B, Jia S, Bian X, Zhao K, Li J, Liu P, Li J. Chromosome-level genome of the long-tailed marine-living ornate spiny lobster, Panulirus ornatus. Sci Data. 2024 Jun 22;11(1):662. doi: 10.1038/s41597-024-03512-9.

## Test dataset generation

The following samples were chosen for the test dataset:

| Sample | Phase    | Stage           | Days post-molt | SRX accession | SRR accession |
| ------ | -------- | --------------- | -------------- | ------------- | ------------- |
| H4     | Puerulus | Clear post-molt | 1              | SRX9161265    | SRR12681119   |
| H5     | Puerulus | Clear post-molt | 1              | SRX9161266    | SRR12681118   |
| H6     | Puerulus | Clear post-molt | 1              | SRX9161267    | SRR12681117   |
| K1     | Juvenile | Post-molt       | 0              | SRX9161275    | SRR12681109   |
| K2     | Juvenile | Post-molt       | 0              | SRX9161276    | SRR12681108   |
| K3     | Juvenile | Post-molt       | 0              | SRX9161277    | SRR12681107   |

For the minimal test dataset, 2 000 reads were sampled from each FASTQ file using BBMap's (v39.01) `reformat.sh`. For the full dataset 80 000 reads were sampled from each FASTQ file.

Example of command used for subsampling reads:

```bash
reformat.sh in1=reads_1.fastq.gz in2=reads_2.fastq.gz out1=reads_subset_1.fastq.gz out2= reads_subset_2.fastq.gz sample=2000
```

Assembled transcriptomes generated from the test datasets are also provided, as
they will be useful for testing the pipeline in "quant_only" mode. These
transcriptomes were generated by using the `test` and `test_full` profiles of nf-core/denovotranscript pipeline v1.1.0.
